utils.pwsh/Setup-Target.ps1

```diff
@@ -77,29 +77,15 @@ function Find-VisualStudio {
             Find-VisualStudio
     #>
 
-    $VisualStudioData = Get-CimInstance MSFT_VSInstance
-
-    # Prefer VS versions in this order:
-    # 1. VS2022 Release (stable)
-    # 2. VS2022 Preview
-    # 3. VS2019 Release
-    [string[]]$SupportedVSVersions =
-        "VisualStudio.17.Release",
-        "VisualStudio.17.Preview",
-        "VisualStudio.16.Release"
-    $NumSupportedVSVersions = $SupportedVSVersions.length
-
-    if ( $VisualStudioData.GetType() -eq [object[]] ) {
-        for ( $i = 0; $i -lt $NumSupportedVSVersions; $i++ ) {
-            $VisualStudioDataTemp = ($VisualStudioData | Where-Object {$_.ChannelId -eq $SupportedVSVersions[$i]} | Sort-Object -Property Version)[0]
-            if ( $VisualStudioDataTemp ) {
-                break;
-            }
-        }
-        $VisualStudioData = $VisualStudioDataTemp
+   $VSSetupData = Get-InstalledModule VSSetup -ErrorAction SilentlyContinue
+
+    if ( $VSSetupData -eq $null ) {
+        Install-Module VSSetup -Scope CurrentUser
```

Comment:
Are there any permissions required to install this module in this scope? Are there any known reasons it could fail to install? Should we handle a failure case, or will the script bail out on its own?

Reply:
None listed by Microsoft - it might prompt for installation permission when run on a local system, but that'd be something I would leave to a user by design and not override.

Reply:
Sounds good to me.

Reply:
As an aside, this will probably cause a user prompt, which is fine, but something to be aware of.
```powershell
PS C:\> Install-Module VSSetup -Scope CurrentUser

Untrusted repository
You are installing the modules from an untrusted repository. If you trust this repository, change its
InstallationPolicy value by running the Set-PSRepository cmdlet. Are you sure you want to install the modules from
'PSGallery'?
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "N"):
```

Reply:
Correct. Personally I feel it's a good choice not to suppress this, but let users handle this on their local machines. It's in any case simpler than whatever script switches we could provide.